### PR TITLE
Add a simple votes controller

### PR DESCRIPTION
**Pull request by pezholio**
## Changes
- Users can rate a place
- After rating, they get redirected to the home page where a new place is selected
- The places get randomly selected from the pool of places that the user hasn't rated already
- The user is identified by a session cookie

Screenshot 2022-04-19 at 18 45 44

### CODE REVIEW

**Comment by CristinaRO** [comment]
[Users can submit a vote on a place](https://trello.com/c/dDRBY8RV/16-users-can-submit-a-vote-on-a-place)

File: spec/features/user_rates_a_place_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+RSpec.describe "A user rates a place" do
+  scenario "a successful vote" do
+    # Given that at least a place exists in the database
```
Comment (pezholio) [inline spec/features/user_rates_a_place_spec.rb]: Oh, I like the idea of having Gherkin-style comments 👍

File: app/controllers/places_controller.rb [hunk 2]
```diff
@@ -2,7 +2,16 @@ class PlacesController < ActionController::Base
   layout "application"
 
   def show
-    @place = Place.order(Arel.sql("RANDOM()")).first
-    @vote = @place.votes.new
+    @place = Place.left_joins(:votes)
+      .where("votes.id IS NULL OR votes.uuid != '#{uuid}'")
+      .order(Arel.sql("RANDOM()"))
+      .first
```
Comment (pezholio) [inline app/controllers/places_controller.rb:8]: Could we just return a random place if the `@place` is nil?

**Review by pezholio — COMMENTED** [review]
Can't approve this, because I opened it, but LGTM 👍

**Comment by pezholio** [comment]
@CristinaRO I'm gonna close this. Could you recreate it please, so I can approve?

**Pull request closed without merging**